python/ray/data/_internal/arrow_serialization.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+
+def _register_arrow_json_readoptions_serializer():
+    import ray
+
+    if (
+        os.environ.get(
+            "RAY_DISABLE_CUSTOM_ARROW_JSON_OPTIONS_SERIALIZATION",
+            "0",
+        )
+        == "1"
+    ):
+        import logging
+
+        logger = logging.getLogger(__name__)
+        logger.info("Disabling custom Arrow JSON ReadOptions serialization.")
+        return
+
+    try:
+        import pyarrow.json as pajson
+    except ModuleNotFoundError:
```

Comment:
Should we do this import guard in other files too?

Reply:
Only needed here since this runs at `ray.data` import time, all other such imports should fail since Arrow will be hard requirement for those code paths.